src/Features/LanguageServer/Protocol/Api/ILanguageServerFactory.cs

```diff
@@ -0,0 +1,18 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using StreamJsonRpc;
+
+namespace Microsoft.CodeAnalysis.LanguageServer.Api
+{
+    internal interface ILanguageServerFactory
```

Comment:
Main part of the new API - creates an InProcLanguageServer and registers as a target for the passed in jsonrpc object.